cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -820,6 +842,18 @@ protected Future<Boolean> delete(Reconciliation reconciliation) {
                 .map(Boolean.FALSE); // Return FALSE since other resources are still deleted by garbage collection
     }
 
+    /**
+     * Remove the metrics specific to the kind implementing it.
+     *
+     * @param desiredNames  Set of resources which should be reconciled
+     * @param namespace     The namespace to reconcile, or {@code *} to reconcile across all namespaces.
+     */
+    @Override
+    public void removeMetrics(Set<NamespaceAndName> desiredNames, String namespace) {
+        metrics.removeMetricsForCertificates(key ->
+                !desiredNames.contains(new NamespaceAndName(key.getNamespace(), key.getClusterName())));
+    }
```

Comment:
Ok, long story short ... the only thing you should have trusted me in is when I said on the beginning that this issue is much harder than it seems. Sorry 😞.

But this thing should definitely work (I actually tried it out):

```suggestion
    public void removeMetrics(Set<NamespaceAndName> desiredNames, String namespace) {
        if (ConfigParameter.ANY_NAMESPACE.equals(namespace)) {
            metrics.removeMetricsForCertificates(key ->
                    // When watching all namespaces, we remove all metrics that do not belong to existing clusters
                    !desiredNames.contains(new NamespaceAndName(key.getNamespace(), key.getClusterName())));
        } else {
            metrics.removeMetricsForCertificates(key ->
                    // When watching only one namespace, we remove all metrics that belong to our namespace but not to an existing cluster
                    // We ignore the metrics from other namespaces
                    namespace.equals(key.getNamespace()) && !desiredNames.contains(new NamespaceAndName(key.getNamespace(), key.getClusterName())));
        }
    }
```

Reply:
Trust me, I do trust you 😄 
We're really keen to solve this!

Thanks for all your help!